extract-metadata.py

```diff
@@ -0,0 +1,84 @@
+from dandi.dandiapi import DandiAPIClient
+import pandas as pd
+from pathlib import Path
+
+modalities = {'oct': 'PS-OCT',
+              'df': 'Dark Field Microscopy',
+              'xpct': 'HiP-CT',
+              'dwi': 'DWI',
+              'fluo': 'LSM',
+              'photo': 'Blockface photo'}
+
+# Create dataframe of all assets across all datasets
+def extract_assets():
+    client = DandiAPIClient("https://api.lincbrain.org/api")
+    client.dandi_authenticate()
+    dandisets = client.get_dandisets()
+
+    print(f"Processing {sum(1 for _ in dandisets)} Dandisets on lincbrain.org")
+
+    df = pd.DataFrame(columns=["Dandiset",
+                            "Version",
+                            "Subject", 
+                            "Modality",
+                            "Path",
+                            "Filename",
+                            "Extension",
+                            "Directory", # Top-level directory (e.g. source data, raw data, derivatives)
+                            'Size (bytes)'])
+
+    for dandiset in dandisets:
+        latest_dandiset = dandiset.for_version('draft')
+        for asset in latest_dandiset.get_assets():
+            print(f"Dandiset: {latest_dandiset}; Asset: {asset.path.split('/')[-1]:<40}", end='\r')
+
+            metadata = asset.get_metadata()
+            metadata_dict = metadata.model_dump(mode='json', exclude_none=True)
+
+            subject = 'Unknown'
+            for part in asset.path.split('/'):
+                if part.startswith("sub-"):
+                    subject = part.split("sub-")[1].split('_')[0]
+                    break
+
+            if subject == 'Unknown' and any(filename in asset.path.split('/')[-1].lower() for filename in ['dataset_description.json', 'participants.tsv', 'readme.md', 'samples.tsv']):
+                subject = 'n/a'
+
+            modality = next((value for key, value in modalities.items() 
+                            if key in asset.path.split('/')[-1].lower()), 
```

Comment:
You call `asset.path.split` a few times here -- you could probably make a mini-improvement by storing the values of the split in one variable, then evaluating each part of the split wherever you call `asset.path.split` again

Reply:
Good idea.  Just updated.   Thanks.